test/system_tests/test_coupled_heat_hydrogen.py

```diff
@@ -0,0 +1,239 @@
+            * (n_trap - (exact_trapped_solution(x, t)))
+            - p(x, t) * (exact_trapped_solution(x, t))
+        )
+
+    def mms_trapped_source(x, t):
+        return (
+            dtrappeddt
+            + k(x, t)
+            * (exact_mobile_solution(x, t))
+            * (n_trap - (exact_trapped_solution(x, t)))
+            - p(x, t) * (exact_trapped_solution(x, t))
+        )
+
+    test_hydrogen_problem = F.HydrogenTransportProblem(
+        mesh=test_mesh,
+        subdomains=[test_vol_sub, left_sub, right_sub],
+        boundary_conditions=[
+            F.FixedConcentrationBC(
+                subdomain=left_sub,
+                value=exact_mobile_solution,
+                species=test_mobile,
+            ),
+            F.FixedConcentrationBC(
+                subdomain=right_sub,
+                value=exact_mobile_solution,
+                species=test_mobile,
+            ),
+            F.FixedConcentrationBC(
+                subdomain=left_sub,
+                value=exact_trapped_solution,
+                species=test_trapped,
+            ),
+            F.FixedConcentrationBC(
+                subdomain=right_sub,
+                value=exact_trapped_solution,
+                species=test_trapped,
+            ),
+        ],
+        species=[test_mobile, test_trapped],
+        reactions=[
+            F.Reaction(
+                reactant=[test_mobile, test_traps],
+                product=test_trapped,
+                k_0=k_0,
+                E_k=E_k,
+                p_0=p_0,
+                E_p=E_p,
+                volume=test_vol_sub,
+            )
+        ],
+        initial_conditions=[
+            F.InitialCondition(value=exact_mobile_intial_cond, species=test_mobile),
+            F.InitialCondition(value=exact_trapped_intial_cond, species=test_trapped),
+        ],
+        sources=[
+            F.ParticleSource(
+                value=mms_mobile_source, volume=test_vol_sub, species=test_mobile
+            ),
+            F.ParticleSource(
+                value=mms_trapped_source, volume=test_vol_sub, species=test_trapped
+            ),
+        ],
+        settings=F.Settings(
+            atol=1e-10,
+            rtol=1e-10,
+            transient=True,
+            stepsize=final_time / 20,
+            final_time=final_time,
+        ),
+    )
+
+    # define coupled problem
+    test_coupled_problem = F.CoupledtTransientHeatTransferHydrogenTransport(
+        heat_problem=test_heat_problem,
+        hydrogen_problem=test_hydrogen_problem,
+    )
+    test_coupled_problem.initialise()
+    test_coupled_problem.run()
+
+    # compare computed values with exact solutions
+    T_computed = test_coupled_problem.hydrogen_problem.temperature_fenics
+    mobile_computed = test_mobile.post_processing_solution
+    trapped_computed = test_trapped.post_processing_solution
+
+    exact_T = lambda x: 3 * x[0] ** 2 + 10 * final_time
+    exact_mobile = lambda x: 2 * x[0] ** 2 + 15 * final_time
```

Comment:
```suggestion
    exact_mobile = lambda x: exact_mobile_solution(x, t=final_time)
```

same for the two others

Reply:
You didn't do it for the two others